plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/DeployCommandHandler.java

```diff
@@ -166,7 +167,7 @@ public void done(IJobChangeEvent event) {
     deploy.schedule();
   }
 
-  protected abstract StagingDelegate getStagingDelegate(IProject project);
+  protected abstract StagingDelegate getStagingDelegate(IProject project) throws CoreException;
```

Comment:
I'm surprised this didn't already throw an exception. Were we missing a runtime exception somewhere?

Reply:
`CoreException` is a checked exception that was not needed before. It is now thrown from `ProjectFacetsManager.create()` (e.g., when a project is closed) introduced in this PR, so I decided to propagate.

Reply:
I think it was needed. We were just throwing the wrong (unchecked) exception instead.